### ISSUE TITLE
quarkus-cache in dev mode ends up with NPE

### ISSUE THREAD

**Issue by rsvoboda** [quarkusio/quarkus#14439]
Bootstrapped project with quarkus-cache ends up with NPE in dev mode.
`mvn test` execution works well

```
wget -q -O xx.zip https://code.quarkus.io/api/download?s=W1i
unzip -q xx.zip
cd code-with-quarkus
mvn quarkus:dev
```
```
[INFO] Scanning for projects...
[INFO]
[INFO] ---------------------< org.acme:code-with-quarkus >---------------------
[INFO] Building code-with-quarkus 1.0.0-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO]
[INFO] --- quarkus-maven-plugin:1.11.0.Final:dev (default-cli) @ code-with-quarkus ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 2 resources
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 1 source file to /Users/rsvoboda/tmp/code-with-quarkus/code-with-quarkus/target/classes
Listening for transport dt_socket at address: 5005
__  ____  __  _____   ___  __ ____  ______
 --/ __ \/ / / / _ | / _ \/ //_/ / / / __/
 -/ /_/ / /_/ / __ |/ , _/ ,< / /_/ /\ \
--\___\_\____/_/ |_/_/|_/_/|_|\____/___/
2021-01-20 16:41:40,360 INFO  [io.qua.dep.dev.IsolatedDevModeMain] (main) Attempting to start hot replacement endpoint to recover from previous Quarkus startup failure
2021-01-20 16:41:40,359 ERROR [io.qua.run.Application] (Quarkus Main Thread) Failed to start application (with profile dev): java.lang.NullPointerException: Cannot invoke "java.util.Collection.size()" because "names" is null
	at io.quarkus.cache.runtime.CaffeineCacheSupplier.get(CaffeineCacheSupplier.java:21)
	at io.quarkus.cache.runtime.CaffeineCacheSupplier.get(CaffeineCacheSupplier.java:15)
	at io.quarkus.vertx.http.runtime.devmode.DevConsoleRecorder.addInfo(DevConsoleRecorder.java:37)
	at io.quarkus.deployment.steps.DevConsoleProcessor$runtimeTemplates-308161071.deploy_0(DevConsoleProcessor$runtimeTemplates-308161071.zig:95)
	at io.quarkus.deployment.steps.DevConsoleProcessor$runtimeTemplates-308161071.deploy(DevConsoleProcessor$runtimeTemplates-308161071.zig:40)
	at io.quarkus.runner.ApplicationImpl.doStart(ApplicationImpl.zig:631)
	at io.quarkus.runtime.Application.start(Application.java:90)
	at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:97)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:66)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:42)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:119)
	at io.quarkus.runner.GeneratedMain.main(GeneratedMain.zig:29)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:64)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:564)
	at io.quarkus.runner.bootstrap.StartupActionImpl$3.run(StartupActionImpl.java:134)
	at java.base/java.lang.Thread.run(Thread.java:832)

Quarkus application exited with code 1
Press Enter to restart or Ctrl + C to quit
```

**Comment by gsmet** [comment]
Can you try with master? I think I already fixed that one.

**Comment by gsmet** [comment]
Or even better, try the 1.11 branch as the fix should be there.

**Comment by rsvoboda** [comment]
Building 1.11 branch

**Comment by gwenneg** [comment]
Same issue as #14436.

**Comment by rsvoboda** [comment]
The issue is gone when using bits from 1.11 branch.
Probably fixed by @gsmet in https://github.com/quarkusio/quarkus/pull/14376/commits/c12e5838fa5657c13c740987b3116c421acae8bb

**Comment by gwenneg** [comment]
That is correct @rsvoboda. The fix is already on the `1.11` branch but is was backported after the `1.11.0.Final` tag so it will probably be included in the `1.11.1` release.

**Comment by gwenneg** [comment]
The workaround I posted in #14436 should work with `1.11.0.Final`.
